pints/_mcmc/_adaptive.py

```diff
@@ -26,27 +28,190 @@ class AdaptiveCovarianceMCMC(pints.MCMC):
+        burn-in.
+        """
+        return self._burn_in_rate
+
+    def thinning(self):
+        """
+        Return thinning.
+        """
+        return self._thinning
+
+    def set_acceptance_rate(self, rate):
+        """
+        Set target acceptance rate.
+        """
+        self._acceptance_target = rate
+
+    def set_iterations(self, iterations):
+        """
+        Set the total number of iterations.
+        """
+        self._iterations = iterations
+        self._adaptation = int(self._iterations * self._adaptation_rate)
+        self._burn_in = int(self._iterations * self._burn_in_rate)
+        if self._thinning > self._adaptation:
+            print('WARNING: Thinning larger than adaptation')
+        if self._thinning > self._burn_in:
+            print('WARNING: Thinning larger than burn-in')
+        if self._thinning > self._iterations:
+            raise Exception('Thinning larger than total number of iterations')
+
+    def set_adaptation_rate(self, rate):
+        """
+        Set the first fraction of the total number of iterations before 
+        using adaptation.
+        """
+        self._adaptation_rate = rate
+        self._adaptation = int(self._iterations * self._adaptation_rate)
+        if self._adaptation > self._burn_in:
+            print('WARNING: Adaptation use after burn-in')
+
+    def set_adaptation(self, adaptation):
+        """
+        Set the number of iterations before using adapation.
+        """
+        self._adaptation = adaptation
+        self._adaptation_rate = self._adaptation/self._iterations
+        if self._adaptation > self._burn_in:
+            print('WARNING: Adaptation use after burn-in')
+
+    def set_burn_in_rate(self, rate):
+        """
+        Set the first fraction of the total number of iterations to discard as 
+        burn-in.
+        """
+        self._burn_in_rate = rate
+        self._burn_in = int(self._iterations * self._burn_in_rate)
+        if self._adaptation > self._burn_in:
+            print('WARNING: Adaptation use after burn-in')
+
+    def set_burn_in(self, burn_in):
+        """
+        Set the number of iterations to discard as burn-in.
+        """
+        self._burn_in = burn_in
+        self._burn_in_rate = self._burn_in/self._iterations
+        if self._adaptation > self._burn_in:
+            print('WARNING: Adaptation use after burn-in')
+
+    def set_thinning(self, thinning):
+        """
+        Set thinning - store only one sample per X
+        """
+        self._thinning = thinning
+        if self._thinning > self._adaptation:
+            print('WARNING: Thinning larger than adaptation')
+        if self._thinning > self._burn_in:
+            print('WARNING: Thinning larger than burn-in')
+        if self._thinning > self._iterations:
+            raise Exception('Thinning larger than total number of iterations')
+
+    def print_setup(self):
```

Comment:
This is something the user can do her/himself!

Remember if we add these methods, and people use them, then we won't be able to remove them anymore without breaking their code, and then we're stuck with a whole bunch of methods that were convenient for 1 specific project for the rest of Pints' lifetime!